File: economia_computazionale/lezioni/02/script/codice.m
c=0.63;
b=1500;
k=0.6;
h=2700;

C_bar=55;
I_bar=75;
G_bar=200;
T_bar=110;
M_bar=200;
P_bar=1;

xNames=["Y";"C";"I";"R"]; 
A=[
1 -1 -1 0; 
-c 1 0 0;
0 0 1 b; 
k 0 0 -h
];
d=[G_bar;C_bar-c*T_bar;I_bar;M_bar/P_bar];

[L,U]=lu(A);
y=L\d;
x=U\y;
stamp=[xNames num2str(round(x*10000)/10000)];
disp(cstrcat(stamp(1:4,1),repmat(repelem(" ",4)',1,3),stamp(1:4,2:9)));

File: economia_computazionale/lezioni/02/script/codice.py
c=0.63
b=1500
k=0.6
h=2700

C_bar=55
I_bar=75
G_bar=200
T_bar=110
M_bar=200
P_bar=1

import numpy as np
xNames=np.array(["Y","C","I","R"]).reshape(4,1) 
A=[
[1,-1,-1,0],
[-c,1,0,0],
[0,0,1,b],
[k,0,0,-h]
]
d=[G_bar,C_bar-c*T_bar,I_bar,M_bar/P_bar]

import scipy as sp

P,L,U=sp.linalg.lu(A)
x=sp.linalg.solve(A,d)
x1=np.array(x.round(4)).reshape(4,1)

toShow=np.hstack([xNames,x1])

for i in range(4):
    print(toShow[i][0],"  ",toShow[i][1])


